src/main/java/patio/voting/services/internal/DefaultVotingService.java

```diff
@@ -144,7 +146,11 @@ private Voting createVotingIfSuccess(Optional<UserGroup> userGroup) {
         .thenCheck(() -> userIsInGroup.check(user, group))
         .thenCheck(() -> anonymousAllowed.check(input.isAnonymous(), isGroupAnonymous))
         .then(createVote(voting, user, input))
-        .sideEffect(v -> updateVotingAverage(v.getVoting()));
+        .sideEffect(
+            (v) -> {
+              updateVotingAverage(v.getVoting());
```

Comment:
Aren't these two statements updating the same `VotingStats` instance ? I think the sole responsibility of updating the stats belongs to the `VotingStatService`. Updating average should be moved to VotingStatsService too.

Reply:
You're right, I guess it sould also be its responsability now. I'll refactor this